lib/Runtime/Library/JsBuiltIn/JsBuiltIn.js

```diff
@@ -238,5 +241,157 @@
 
         return a;
     });
-  
+
+    platform.registerChakraLibraryFunction("FlattenIntoArray", function(target, source, sourceLen, start, depth)
+    {
+        "use strict";
+        //1. Let targetIndex be start.
+        let targetIndex = start;
+        //2. Let sourceIndex be 0.
+        let sourceIndex = 0;
+        //3. Repeat, while sourceIndex < sourceLen
+        let element;
+        while (sourceIndex < sourceLen) {
+            // a. Let P be ! ToString(sourceIndex).
+            // b. Let exists be ? HasProperty(source, P).
+            if (sourceIndex in source) {
+                // c. If exists is true, then
+                //  i. Let element be ? Get(source, P).
```

Comment:
Does the code as written work correctly in the presence of proxies / getters? The spec seems to say that the element is accessed once, while below you access it multiple times.

Reply:
Should now be fixed - I've also added a case using a Proxy to test this.

Reply:
It’d be great to contribute those tests back to test262 :-)

Reply:
@ljharb I'll endeavour to contribute some equivalent tests to test262 later this week - I've not written a test262 test before so will take me a bit of time to work out how to.